### PR TITLE
Get the SSE2 condition match the emmintrin.h inclusion condition.

### PR DESCRIPTION
**Pull request by vrabaud**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Review by asmorkalov — APPROVED** [review]
:+1:

File: modules/core/include/opencv2/core/fast_math.hpp [hunk 1]
```diff
@@ -201,7 +201,7 @@ cvRound( double value )
 {
 #if defined CV_INLINE_ROUND_DBL
     CV_INLINE_ROUND_DBL(value);
-#elif ((defined _MSC_VER && defined _M_X64) || (defined __GNUC__ && defined __SSE2__)) && !defined(__CUDACC__)
+#elif ((defined _MSC_VER && defined _M_X64) || (defined __GNUC__ && defined __x86_64__ && defined __SSE2__)) && !defined(__CUDACC__)
```
Comment (opencv-alalek) [inline modules/core/include/opencv2/core/fast_math.hpp:204]: Condition of include should be changed instead.

Comment (vrabaud) [reply modules/core/include/opencv2/core/fast_math.hpp:204]: I actually was wondering whether it made sense to check for __x86_64__ AND __SSE2__.
Done

**Review by opencv-alalek — APPROVED** [review]
Thank you :+1:

**Pull request merged**